File: src/main/java/org/books/services/BarcodeImageService.java
package org.books.services;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.google.zxing.client.j2se.MatrixToImageWriter;
import com.google.zxing.common.BitMatrix;
import com.google.zxing.oned.EAN13Writer;
import com.google.zxing.qrcode.QRCodeWriter;
import org.springframework.stereotype.Service;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;

@Service
public class BarcodeImageService {

    public InputStream generateQRCode(String text, int width, int height) {
        QRCodeWriter qrCodeWriter = new QRCodeWriter();
        BitMatrix bitMatrix = null;
        try {
            bitMatrix = qrCodeWriter.encode(text, BarcodeFormat.QR_CODE, width, height);
        } catch (WriterException e) {
            e.printStackTrace();
        }

        BufferedImage image = MatrixToImageWriter.toBufferedImage(bitMatrix);

        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        try {
            ImageIO.write(image, "png", baos);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return new ByteArrayInputStream(baos.toByteArray());
    }

    public InputStream generateEAN13Barcode(String text, int width, int height) {
        EAN13Writer ean13Writer = new EAN13Writer();
        BitMatrix bitMatrix = null;
        try {
            bitMatrix = ean13Writer.encode(text, BarcodeFormat.EAN_13, width, height);
        } catch (WriterException e) {
            e.printStackTrace();
        }

        BufferedImage image = MatrixToImageWriter.toBufferedImage(bitMatrix);

        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        try {
            ImageIO.write(image, "png", baos);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return new ByteArrayInputStream(baos.toByteArray());
    }

}

File: src/main/java/org/books/services/BarcodeImageService.kt
package org.books.services

import com.google.zxing.BarcodeFormat
import com.google.zxing.WriterException
import com.google.zxing.client.j2se.MatrixToImageWriter
import com.google.zxing.common.BitMatrix
import com.google.zxing.oned.EAN13Writer
import com.google.zxing.qrcode.QRCodeWriter
import org.springframework.stereotype.Service

import javax.imageio.ImageIO
import java.awt.image.BufferedImage
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.io.InputStream

@Service
class BarcodeImageService {

    fun generateQRCode(text: String, width: Int, height: Int): InputStream {
        val qrCodeWriter = QRCodeWriter()
        var bitMatrix: BitMatrix? = null
        try {
            bitMatrix = qrCodeWriter.encode(text, BarcodeFormat.QR_CODE, width, height)
        } catch (e: WriterException) {
            e.printStackTrace()
        }

        val image = MatrixToImageWriter.toBufferedImage(bitMatrix!!)

        val baos = ByteArrayOutputStream()
        try {
            ImageIO.write(image, "png", baos)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return ByteArrayInputStream(baos.toByteArray())
    }

    fun generateEAN13Barcode(text: String, width: Int, height: Int): InputStream {
        val ean13Writer = EAN13Writer()
        var bitMatrix: BitMatrix? = null
        try {
            bitMatrix = ean13Writer.encode(text, BarcodeFormat.EAN_13, width, height)
        } catch (e: WriterException) {
            e.printStackTrace()
        }

        val image = MatrixToImageWriter.toBufferedImage(bitMatrix!!)

        val baos = ByteArrayOutputStream()
        try {
            ImageIO.write(image, "png", baos)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return ByteArrayInputStream(baos.toByteArray())
    }

}
